File: 2021/day06/python/day06.py

#keeps track of how many fishes there are for any age
def simulate(fishes,part):
    if part==1:
        loops=80
    elif part==2:
        loops=256
    ages = [0]*9
    for i in fishes:
        ages[i] += 1
    for day in range(0, loops):
        cycle = ages[0]
        for age in range(0, 9):
            ages[age-1] = ages[age]
        ages[6] += cycle
        ages[8] = cycle
    print(sum(ages))



with open('../input.txt') as f:
    fishes = [int(x) for x in f.read().split(',')]

simulate(fishes,1)
simulate(fishes,2)

File: 2021/day06/rust/day06.rs
use std::fs;

fn simulate(input: Vec<usize>) {
    
    let mut ages: [usize; 9] = [0; 9];
    for i in input { ages[i] += 1; }

    for day in 0..257 {
        if day == 80 {
            let result: usize=ages.iter().sum();
            println!("{}",result);
        }
        if day == 256 {
            let result: usize=ages.iter().sum();
            println!("{}",result);
            break;
        }
        ages[(day + 7) % 9] += ages[day % 9];
    }
}

fn read_input(filename: &str) -> Vec<usize> {
    fs::read_to_string(filename)
        .expect("Error on reading input file")
        .split(',')
        .map(|n| n.parse::<usize>().unwrap())
        .collect()
}


fn main() {
    let input=read_input("../input.txt");
    simulate(input);
}
